### ISSUE TITLE
Beta testing

### ISSUE THREAD

**Issue by katrina-cityofdetroit** [CityOfDetroit/Departmental-Budget-Request-Form#83]
Publish new version to GH pages and share with Noah

Noah feedback 9/9
- [x] #95
- [x] #96
- [x] #97
- [x] #98

**Comment by katrina-cityofdetroit** [comment]
- We need to include the job codes along with the job titles. There's no way you would've known this, but job titles are not unique and are often shared among different job codes, because of reasons. For example, there are two different job codes with the title "General Auto Mechanic," one code for DPW employees and one code for DDoT employees. They are covered by different unions and therefore have different pay ranges. So, we need to include the job code in there so that we can identify the proper pay range and/or benefits package.
- When entering a new job or new non-personnel item, I wasn't able to enter new Appropriations or Cost Centers. Every new fiscal year inevitably brings a few new approp or cc codes, so they'll need to be able to enter new ones that aren't part of the initial upload.
- I wasn't able to add new jobs because there was nothing in the Service drop-down menu. I know Services are still a work-in-progress, though.
- Also when entering a new job, they shouldn't be entering 'Expected fringe rate.' That's something we (Office of Budget) assign based on the Job Code (it's part of the Gold Book personnel data).
- At the end of the process, it wouldn't let me download the new Excel file; it kept telling me I was above baseline target, even though I brought it below

**Comment by katrina-cityofdetroit** [comment]
Re job codes: Would you prefer that they have a separate column or can I combine them in the job title cells?
- If it's easier to combine them, that's fine
Re new Approp/CC codes: I thought we didn't want to let users create new codes? Or is that only true for revenue? To implement this, is it okay to let the user just type in any number?
- Yeah...this is a tough one. In an ideal world they wouldn't create new codes, but I just know from experience that every fiscal year we end up having new approp or cc codes. I think we're gonna have to let them put in new codes, and the analysts will just have to review them when we get them back. Approps are always five digits and Cost Centers are always six, if that matters.
Services wouldn't show up if they're not in the DS drop-down menu. I can make that field optional to avoid the lock on new positions.
- I think it was an issue for me because I didn't have any services in the drop-down on the sample file I used. But I think we do want to make them put in services for every line. I'll update my sample file and test it again on the next version.
Re fringe rate: would you prefer that I leave this number blank for new jobs in the table/Excel download or try to have the web tool assume it from job code and BU?
- The standard Detail Sheet fills in the Fringe Rate (column AB) via an xlookup based on the Job Code (column N). As long as we fill in the job code in col N, the fringe rate should fill automatically via the xlookup

**Comment by katrina-cityofdetroit** [comment]
- [x] Add job code to personnel page
- [x] Allow user to add new appropriation or cost center code
- [x] Fill in fringe rate automatically using job code
- [x] Fix Excel download block bug

**Comment by katrina-cityofdetroit** [comment]
Questions: 
 - is employee type unique for each job code? Is it different from step/merit group?
 - When entering new job, should we let them enter salary or automatically enter from the center of the Gold Book range?
 - Gold Book range is FY24, is that right?
 -  Should we allow revenue for new initiatives? (No space on DS)
- Why have position status for personnel tab if rolled up?

**Comment by katrina-cityofdetroit** [comment]
Question: Should we allow revenue for new initiatives? (No space on DS)

**Comment by katrina-cityofdetroit** [comment]
## Noah answers to my questions 9/16

### Is employee type unique for each job code? Is it different from step/merit group? Basically, I'm trying to figure out whether I need to ask for employee type when adding a new job or if I can fill that in automatically
- Ugh, I wish it was unique. Most of the time it can be tied to job codes, e.g. Administrative Special Services Staff are always TASS. But there are always exceptions. For example, there are some uniformed officers who have non-police-ish job titles like "Clerk 3 Human Resources," but we have to list them as Uniform Police. So, unfortunately we need to prompt the users to enter the Employee Type.

### When entering new job, should we let them enter salary or automatically enter from the center of the Gold Book range?
- If it's brand new position, we can automatically enter the Gold Book median. New positions should (hypothetically) only appear as part of Supplemental Initiatives. If it's part of their Baseline and they're just updating the data (like changing the cost center), they can use the existing average.
- Btw, per Donnie and Matt we've added "New Position" to the drop-down menu in the 'Position Status' column.

### Should we allow revenue for new initiatives? (There's no column on the DS for this).
- If they're requesting Supplemental Revenue, it should always be on a completely different initiative from any Expenditure initiative. No initiative should ever have both Exp and Rev. That's why we didn't include a Rev column on the Initiatives Summary. We kind of want to discourage any Supplemental Revenue requests, since Rev should be mostly driven by the Revenue Estimating Conference.

### Do you anticipate keeping the 'position status' column on the personnel tab of the detail sheet? If so, how would you interpret 'vacant' or 'filled' for a row with FTE > 1?
- Yes, we're keeping the 'Position Status' column. As I noted earlier, we added "New Position" as an option in the drop-down.
- If there is a job code that has both filled and vacant FTEs, then they should each be entered on separate rows. For example, on my prototype DS, in 28290-290010 there are two lines for 413050 CRIO Specialist; 6.0 FTEs are filled, and 2.0 FTEs are vacant.

**Comment by katrina-cityofdetroit** [comment]
## My follow up questions

#### If it's brand new position, we can automatically enter the Gold Book median. New positions should (hypothetically) only appear as part of Supplemental Initiatives. If it's part of their Baseline and they're just updating the data (like changing the cost center), they can use the existing average.

I want to make sure I'm understanding this correctly.
- Right now I don't let people edit the cost center directly for the personnel section. To change the cost center, they would have to set FTE = 0 in the original cost center and then add a new job row with the new cost center. In this case, should I allow the user to enter their own average salary or set the salary to the average for the same job code in other cost centers? (Or let users edit the cost center?
- In the edge case where a department wants to change a position within the baseline (ex. create a Budget Analyst 3 position and delete a Budget Analyst 2 position). If the newly created position doesn't have an average (because it's the first of its kind for the department), should I let the user enter a proposed salary or use the middle of the Gold Book range?
- Last note on this topic: right now I just have a space for total personnel cost for any supplemental initiatives, but there's no place to add individual job titles, etc for supplementals. (Janani asked me to do it this way so that departments just give a ballpark number for new initiatives). 

#### We kind of want to discourage any Supplemental Revenue requests, since Rev should be mostly driven by the Revenue Estimating Conference.

 - Got it. I'll remove the revenue question from the new initiatives form and continue to prevent departments from adding new baseline revenues.
 
#### As I noted earlier, we added "New Position" as an option in the drop-down.

- Got it. I'll make sure that any new positions have this value in the Excel download.

**Comment by katrina-cityofdetroit** [comment]
Ahh, sorry, I think we were operating under different definitions of "new position." I was thinking in terms of "a position that doesn't exist in the budget yet," while you meant "a new position row on the Personnel tab." Your current method is the correct way to do it. I would say let the user enter their own average salary.
 
I messed around with the tool a bit yesterday. When adding a new line, I'm still not seeing Services populate in the drop-down menu, which is preventing me from adding new items. I think we might also need to let users add a new Fund, the way we did for Approps and Cost Centers.
